tendrl/node_agent/monitoring/collectd/collectors/gluster/utils.py

```diff
@@ -329,26 +328,16 @@ def get_size_MB(size_str):
         return float(size_str.split("PB")[0]) * 1024 * 1024 * 1024
 
 
-def find_brick_host(etcd_client, integration_id, brick_host):
+def find_brick_host(etcd_client, brick_host):
     if etcd_client:
         try:
-            int_id = integration_id
-            _key = "indexes/tags/tendrl/integration/%s" % int_id
-            all_nodes = etcd_client.read(_key).value
-            all_nodes = json.loads(all_nodes)
-            for node_id in all_nodes:
-                fqdn = "/nodes/%s/NodeContext/fqdn" % node_id
-                fqdn = etcd_client.read(fqdn).value
-
-                if brick_host in fqdn:
-                    return fqdn
-
-                ip = "/nodes/%s/NodeContext/ipv4_addr" % node_id
-                ip = etcd_client.read(ip).value
-                if brick_host in ip:
-                    return fqdn
-
+            ip = socket.gethostbyname(brick_host)
+            node_id = etcd_client.read("indexes/ip/%s" % ip).value
+            _key = "/nodes/%s/NodeContext/fqdn" % node_id
+            fqdn = etcd_client.read(_key).value
```

Comment:
before reading fqdn

ensure "node_id" is present at ""indexes/tags/tendrl/integration/:integration_id" , 

otherwise there can be clash between IP address in multi cluster scenario